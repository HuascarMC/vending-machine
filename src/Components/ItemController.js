import React from 'react';
import axios from 'axios';

class ItemController extends React.Component {


  increaseItemQuantity(item) {
    axios.put(`https://vending-machine-server.herokuapp.com/machine/additem/${item}`, { crossdomain: true})
    .then((response) => {
      this.props.getItems();
    })
    .catch(function (error) {
      console.log(error);
    })
  }

  reduceItemQuantity(item) {
    axios.put(`https://vending-machine-server.herokuapp.com/machine/removeitem/${item}`, { crossdomain: true})
    .then((response) => {
      this.props.getItems();
    })
    .catch(function (error) {
      console.log(error);
    })
  }

  render() {

    return(
      <div className="controller-wrapper">
        <ul>
          <li>COKE  { this.props.items.coke }<button onClick={()=> this.increaseItemQuantity("coke") } className="plus">+</button><button onClick={ ()=>this.reduceItemQuantity("coke") } className="minus">-</button></li>
          <li>PEPSI { this.props.items.pepsi }<button onClick={()=> this.increaseItemQuantity("pepsi") } className="plus">+</button><button onClick={ ()=>this.reduceItemQuantity("pepsi") } className="minus">-</button></li>
          <li>SODA  { this.props.items.soda }<button onClick={()=> this.increaseItemQuantity("soda") } className="plus">+</button><button onClick={ ()=>this.reduceItemQuantity("soda") } className="minus">-</button></li>
          <li>WATER { this.props.items.water }<button onClick={()=> this.increaseItemQuantity("water") } className="plus">+</button><button onClick={ ()=>this.reduceItemQuantity("water") } className="minus">-</button></li>
        </ul>
      </div>
    );
  }
}

export default ItemController;
