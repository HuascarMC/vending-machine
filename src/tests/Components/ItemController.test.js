import React from 'react';
import ItemController from '../../Components/ItemController'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

var state = {
  coke: 0,
  pepsi: 0,
  soda: 0,
  water: 0
}

it('should render without crashing', () => {
  shallow(<ItemController items={state}/>);
});

it('should render controller-wrapper div', () => {
  const wrapper = shallow(<ItemController items={state}/>);
  const div = wrapper.find('.controller-wrapper');
  expect(div.exists()).toEqual(true);
});

it('should render u-list', () => {
  const wrapper = shallow(<ItemController items={state}/>);
  const ul = wrapper.find('ul');
  expect(ul.exists()).toEqual(true);
});

it('should render four list tags', () => {
  const wrapper = shallow(<ItemController items={state}/>);
  const li = wrapper.find('li');
  expect((li).length).toBe(4);
});

it('should render coke quantity passed by props', () => {
  const wrapper = shallow(<ItemController items={state}/>);
  const li = wrapper.find('li').at(0);
  expect(li.text()).toEqual("COKE  0+-");
});

it('should render pepsi quantity passed by props', () => {
  const wrapper = shallow(<ItemController items={state}/>);
  const li = wrapper.find('li').at(1);
  expect(li.text()).toEqual("PEPSI 0+-");
});


it('should render soda quantity passed by props', () => {
  const wrapper = shallow(<ItemController items={state}/>);
  const li = wrapper.find('li').at(2);
  expect(li.text()).toEqual("SODA  0+-");
});

it('should render water quantity passed by props', () => {
  const wrapper = shallow(<ItemController items={state}/>);
  const li = wrapper.find('li').at(3);
  expect(li.text()).toEqual("WATER 0+-");
});
