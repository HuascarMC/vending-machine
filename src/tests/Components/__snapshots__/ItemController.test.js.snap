// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="controller-wrapper"
>
  <ul>
    <li>
      COKE  
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
    <li>
      PEPSI 
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
    <li>
      SODA  
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
    <li>
      WATER 
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
  </ul>
</div>
`;
